src/librados/RadosClient.cc

```diff
@@ -70,17 +71,16 @@ librados::RadosClient::RadosClient(CephContext *cct_)
   : Dispatcher(cct_->get()),
     cct_deleter{cct_, [](CephContext *p) {p->put();}},
     conf(cct_->_conf),
+    poolctx(cct, ceph::construct_suspended),
     state(DISCONNECTED),
-    monclient(cct_),
+    monclient(cct_, poolctx),
     mgrclient(cct_, nullptr),
     messenger(NULL),
     instance_id(0),
     objecter(NULL),
     lock("librados::RadosClient::lock"),
-    timer(cct, lock),
     refcnt(1),
-    log_last_version(0), log_cb(NULL), log_cb2(NULL), log_cb_arg(NULL),
-    finisher(cct, "radosclient", "fn-radosclient")
+    log_last_version(0), log_cb(NULL), log_cb2(NULL), log_cb_arg(NULL)
```

Comment:
I prefer nullptr to NULL here if possible. YYMV.

Reply:
Done!